src/otx/algorithms/classification/adapters/mmcls/apis/train.py

```diff
@@ -0,0 +1,184 @@
+
+    # The default loader config
+    loader_cfg = dict(
+        # cfg.gpus will be ignored if distributed
+        num_gpus=cfg.ipu_replicas if device == "ipu" else len(cfg.gpu_ids),
+        dist=distributed,
+        round_up=True,
+        seed=cfg.get("seed"),
+        sampler_cfg=cfg.get("sampler", None),
+    )
+    # The overall dataloader settings
+    loader_cfg.update(
+        {
+            k: v
+            for k, v in cfg.data.items()
+            if k not in ["train", "val", "test", "train_dataloader", "val_dataloader", "test_dataloader"]
+        }
+    )
+    # The specific dataloader settings
+    train_loader_cfg = {**loader_cfg, **cfg.data.get("train_dataloader", {})}
+
+    data_loaders = [build_dataloader(ds, **train_loader_cfg) for ds in dataset]
+
+    # put model on gpus
+    if distributed:
+        find_unused_parameters = cfg.get("find_unused_parameters", False)
+        # Sets the `find_unused_parameters` parameter in
+        # torch.nn.parallel.DistributedDataParallel
+        model = wrap_distributed_model(
+            model, cfg.device, broadcast_buffers=False, find_unused_parameters=find_unused_parameters
+        )
+    elif cfg.device == "hpu":
+        assert len(cfg.gpu_ids) == 1
+        model.to(f"hpu:{cfg.gpu_ids[0]}")
+        is_autocast = bool(cfg.get("fp16", False))
+        model = HPUDataParallel(model, dim=0, device_ids=cfg.gpu_ids, is_autocast=is_autocast)
+    else:
+        model = wrap_non_distributed_model(model, cfg.device, device_ids=cfg.gpu_ids)
+
+    # build runner
+    if cfg.device == "hpu":
+        from otx.algorithms.classification.adapters.mmcls.optimizer.hpu_optimizers import register_habana_optimizers
+        habana_optimizers = register_habana_optimizers()
+        if (new_type := "Fused" + cfg.optimizer.get("type", "SGD")) in habana_optimizers:
+            cfg.optimizer["type"] = new_type
+    
+    optimizer = build_optimizer(model, cfg.optimizer)
+
+    if cfg.get("runner") is None:
+        cfg.runner = {"type": "EpochBasedRunner", "max_epochs": cfg.total_epochs}
+        warnings.warn(
+            "config is now expected to have a `runner` section, " "please set `runner` in your config.", UserWarning
+        )
+
+    if device == "ipu":
+        if not cfg.runner["type"].startswith("IPU"):
+            cfg.runner["type"] = "IPU" + cfg.runner["type"]
+        if "options_cfg" not in cfg.runner:
+            cfg.runner["options_cfg"] = {}
+        cfg.runner["options_cfg"]["replicationFactor"] = cfg.ipu_replicas
+        cfg.runner["fp16_cfg"] = cfg.get("fp16", None)
+
+    runner = build_runner(
+        cfg.runner,
+        default_args=dict(
+            model=model, batch_processor=None, optimizer=optimizer, work_dir=cfg.work_dir, logger=logger, meta=meta
+        ),
+    )
+
+    # an ugly walkaround to make the .log and .log.json filenames the same
+    runner.timestamp = timestamp
+
+    # fp16 setting
+    fp16_cfg = cfg.get("fp16", None)
```

Comment:
This can be removed if we do corresponding work in configurer

Reply:
Done.